reconcile/vpc_peerings_validator.py

```diff
@@ -14,10 +16,145 @@
+                            "vpc_peering": {
+                                "vpc_name": mesh_result["vpc_id"],
+                                "cidr_block": mesh_result["cidr_block"],
+                            },
+                        }
+                        create_vpc_peering_dict(
+                            vpc_peering_info,
+                            peerings_enteries_dict,
+                            cluster.name,
+                            cluster.network.vpc,  # type: ignore[union-attr]
+                        )
+                if peering.provider == "account-vpc":
+                    cidr_block = str(peering.vpc.cidr_block)  # type: ignore[union-attr]
+                    vpc_peering_info = {
+                        "provider": peering.provider,
+                        "vpc_peering": {
+                            "vpc_name": peering.vpc.name,  # type: ignore[union-attr]
+                            "region": peering.vpc.region,  # type: ignore[union-attr]
+                            "cidr_block": cidr_block,
+                        },
+                    }
+                    create_vpc_peering_dict(
+                        vpc_peering_info,
+                        peerings_enteries_dict,
+                        cluster.name,
+                        cluster.network.vpc,  # type: ignore[union-attr]
+                    )
+                if peering.provider in (
+                    "cluster-vpc-requester",
+                    "cluster-vpc-accepter",
+                ):
+                    vpc_peering_info = {
+                        "provider": peering.provider,
+                        "vpc_peering": {
+                            "vpc_name": peering.cluster.name,  # type: ignore[union-attr]
+                            "cidr_block": peering.cluster.network.vpc,  # type: ignore[union-attr]
+                        },
+                    }
+                    create_vpc_peering_dict(
+                        vpc_peering_info,
+                        peerings_enteries_dict,
+                        cluster.name,
+                        cluster.network.vpc,  # type: ignore[union-attr]
+                    )
+        overlaps_peering_entries_dict = find_cidr_duplicates_and_overlap(
+            peerings_enteries_dict
+        )
+        if overlaps_peering_entries_dict is False:
+            return False
+    return True
+
+
+def create_vpc_peering_dict(
+    vpc_peering_info, peerings_enteries_dict, cluster_name, cluster_cidr
+):
+    if cluster_name not in peerings_enteries_dict:
+        peerings_enteries_dict[cluster_name] = {
+            "cidr_block": cluster_cidr,
+            "providers": [],
+        }
+    peerings_enteries_dict[cluster_name]["providers"].append(vpc_peering_info)
+
+
+def find_cidr_duplicates_and_overlap(input_dict: dict):
+    items_for_dict = list(input_dict.items())
+    for cluster_name, cluster_data in items_for_dict:
```

Comment:
Since peerings_enteries_dict only contains vpc info for one cluster now, we can simplify it form 
```
 {
        "cluster-test1": {
            "cidr_block": "10.106.0.0/16",
            "providers": [
            ...
            ],
        },
    }
```
to just
```
{
    "cidr_block": "10.25.0.0/16",
    "providers": [
      ....
     ],
 }
```
so we don't need to loop cluster_data here. we can get cluster_name from func args, e.g find_cidr_duplicates_and_overlap(cluster_name: str, input_dict: dict):